drivers/gpio/gpio_kb1200.c

```diff
@@ -0,0 +1,216 @@
+		gpio_regs->GPIOIExx[portnum] |= pinbit;
+		if ((flags & GPIO_SINGLE_ENDED) != 0) {
+			if (flags & GPIO_LINE_OPEN_DRAIN) {
+				gpio_regs->GPIOODxx[portnum] |= pinbit;
+			}
+		} else {
+			gpio_regs->GPIOODxx[portnum] &= (~pinbit);
+		}
+		if (flags & GPIO_PULL_UP) {
+			gpio_regs->GPIOPUxx[portnum] |= pinbit;
+		} else {
+			gpio_regs->GPIOPUxx[portnum] &= (~pinbit);
+		}
+		if ((flags & GPIO_OUTPUT_INIT_HIGH) != 0) {
+			gpio_regs->GPIODxx[portnum] |= pinbit;
+		} else if ((flags & GPIO_OUTPUT_INIT_LOW) != 0) {
+			gpio_regs->GPIODxx[portnum] &= (~pinbit);
+		}
+		gpio_regs->GPIOOExx[portnum] |= pinbit;
+	} else {
+		gpio_regs->GPIOOExx[portnum] &= (~pinbit);
+		if (flags & GPIO_PULL_UP) {
+			gpio_regs->GPIOPUxx[portnum] |= pinbit;
+		} else {
+			gpio_regs->GPIOPUxx[portnum] &= (~pinbit);
+		}
+		gpio_regs->GPIOIExx[portnum] |= pinbit;
+	}
+	return 0;
+}
+
+static int kb1200_gpio_port_get_raw(const struct device *dev, gpio_port_value_t *value)
+{
+	const struct gpio_kb1200_config *config = dev->config;
+	GPIO_T *gpio_regs = GPIO_REG_BASE;
+	*value = gpio_regs->GPIOINxx[config->port_num];
+	return 0;
+}
+
+static int kb1200_gpio_port_set_masked_raw(const struct device *dev, gpio_port_pins_t mask,
+					   gpio_port_value_t value)
+{
+	const struct gpio_kb1200_config *config = dev->config;
+	GPIO_T *gpio_regs = GPIO_REG_BASE;
+	gpio_regs->GPIODxx[config->port_num] |= (value & mask);
+	return 0;
+}
+
+static int kb1200_gpio_port_set_bits_raw(const struct device *dev, gpio_port_pins_t pins)
+{
+	const struct gpio_kb1200_config *config = dev->config;
+	GPIO_T *gpio_regs = GPIO_REG_BASE;
+	gpio_regs->GPIODxx[config->port_num] |= pins;
+	return 0;
+}
+
+static int kb1200_gpio_port_clear_bits_raw(const struct device *dev, gpio_port_pins_t pins)
+{
+	const struct gpio_kb1200_config *config = dev->config;
+	GPIO_T *gpio_regs = GPIO_REG_BASE;
+	gpio_regs->GPIODxx[config->port_num] &= ~pins;
+	return 0;
+}
+
+static int kb1200_gpio_port_toggle_bits(const struct device *dev, gpio_port_pins_t pins)
+{
+	const struct gpio_kb1200_config *config = dev->config;
+	GPIO_T *gpio_regs = GPIO_REG_BASE;
+	gpio_regs->GPIODxx[config->port_num] ^= pins;
+	return 0;
+}
+
+static int kb1200_gpio_pin_interrupt_configure(const struct device *dev, gpio_pin_t pin,
+					       enum gpio_int_mode mode, enum gpio_int_trig trig)
+{
+	const struct gpio_kb1200_config *config = dev->config;
+	GPTD_T *gptd_regs = GPTD_REG_BASE;
+	uint32_t pinbit = BIT(pin);
+	uint32_t portnum = config->port_num;
+
+	/* Check if GPIO port needs interrupt support */
+	if ((mode & GPIO_INT_DISABLE) || (mode & GPIO_INT_ENABLE) == 0) {
+		/* Set the mask to disable the interrupt */
+		gptd_regs->GPTDIExx[portnum] &= (~pinbit);
+	} else {
+		if (mode & GPIO_INT_EDGE) {
+			gptd_regs->GPTDELxx[portnum] &= (~pinbit);
+			if (trig & GPIO_INT_HIGH_1) {
+				if (trig & GPIO_INT_LOW_0) { /* Falling & Rising edge trigger */
+					/* Enable toggle triggeer */
+					gptd_regs->GPTDCHGxx[portnum] |= pinbit;
```

Comment:
We have existing macros for these: https://github.com/zephyrproject-rtos/zephyr/blob/main/include/zephyr/sys/util_macro.h#L61